### DISCUSSION TITLE
Can pytest 7.4.2 run on Python 3.9.4? Startup error with `ExceptionGroup` (introduced in Python 3.11)

### DISCUSSION THREAD

**Discussion by audiodude** [pytest-dev/pytest#11606] [Q&A]
I'm trying to use my pytest testing suite to figure out if my code is compatible with Python 3.9.4.

I have installed Python using asdf and my dependencies using Pipenv, but I get `ModuleNotFoundError` for `exceptiongroup` when I run pytest:

```
tmoney@tmoney-linux:~/code/wp1$ asdf list python
  3.12.0
  3.7.16
 *3.9.4
tmoney@tmoney-linux:~/code/wp1$ python --version
Python 3.9.4
tmoney@tmoney-linux:~/code/wp1$ python3 --version
Python 3.9.4
tmoney@tmoney-linux:~/code/wp1$ pipenv run pip freeze | grep pytest
pytest==7.4.2
pytest-cov==4.1.0
tmoney@tmoney-linux:~/code/wp1$ pipenv run pytest
Traceback (most recent call last):
  File "/home/tmoney/.local/share/virtualenvs/wp1-QZFhY7wp/bin/pytest", line 5, in <module>
    from pytest import console_main
  File "/home/tmoney/.local/share/virtualenvs/wp1-QZFhY7wp/lib/python3.9/site-packages/pytest/__init__.py", line 5, in <module>
    from _pytest._code import ExceptionInfo
  File "/home/tmoney/.local/share/virtualenvs/wp1-QZFhY7wp/lib/python3.9/site-packages/_pytest/_code/__init__.py", line 2, in <module>
    from .code import Code
  File "/home/tmoney/.local/share/virtualenvs/wp1-QZFhY7wp/lib/python3.9/site-packages/_pytest/_code/code.py", line 59, in <module>
    from exceptiongroup import BaseExceptionGroup
ModuleNotFoundError: No module named 'exceptiongroup'
```

The compatibility matrix in the [documentation](https://docs.pytest.org/en/stable/backwards-compatibility.html#python-version-support) claims that pytest 7.1+ runs on Python 3.7+.

**Comment by RonnyPfannschmidt** [comment]
As per the code of the file the exceptiongroup dependency is conditionally imported

As per setup.cfg it should be installed on python 3.9

Please reinstall pytest and check why's the install lacks that

**Comment by audiodude** [comment]
Yes I think pipenv was using a version of pytest that was installed previously when I was accidentally using Python 3.11. I haven't run into this problem again.
